scripts/examples-checker/src/main.rs

```diff
@@ -0,0 +1,168 @@
+    #[clap(long = "paths", short = 'p', multiple_values = true)]
+    pub paths: Vec<PathBuf>,
+
+    /// Targets all Sway examples under /examples
+    #[clap(long = "all-examples")]
+    pub all_examples: bool,
+
+    /// Forc command to run on examples
+    #[clap(arg_enum)]
+    command_kind: CommandKind,
+}
+
+#[derive(ArgEnum, Clone, PartialEq, Eq)]
+enum CommandKind {
+    Build,
+    Fmt,
+}
+
+impl std::fmt::Display for CommandKind {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            CommandKind::Build => write!(f, "build"),
+            CommandKind::Fmt => write!(f, "fmt"),
+        }
+    }
+}
+
+// Check if the given directory contains `Forc.toml` at its root.
+fn dir_contains_forc_manifest(path: &Path) -> bool {
+    if let Ok(entries) = fs::read_dir(path) {
+        for entry in entries.flatten() {
+            if entry.path().file_name().and_then(|s| s.to_str()) == Some("Forc.toml") {
+                return true;
+            }
+        }
+    }
+    false
+}
+
+fn get_sway_path() -> PathBuf {
+    let mut curr_path = std::env::current_dir().unwrap();
+    loop {
+        if curr_path.ends_with("sway") {
+            return curr_path;
+        }
+        curr_path = curr_path.parent().unwrap().to_path_buf()
+    }
+}
+
+/// Returns true if command ran successfully, false otherwise.
+fn run_forc_command(path: &Path, cmd_args: &[&str]) -> bool {
+    if !path.is_dir() || !dir_contains_forc_manifest(path) {
+        return false;
+    }
+
+    let output = std::process::Command::new("forc")
+        .args(cmd_args)
+        .arg(&path)
+        .output()
+        .expect("failed to run command for example project");
+
+    if !output.status.success() {
+        io::stdout().write_all(&output.stdout).unwrap();
+        io::stdout().write_all(&output.stderr).unwrap();
+        false
+    } else {
+        true
+    }
+}
+
+fn run_forc_build(path: &Path) -> bool {
+    run_forc_command(path, &["build", "--path"])
+}
+
+fn run_forc_fmt(path: &Path) -> bool {
+    run_forc_command(path, &["fmt", "--check", "--path"])
+}
+
+fn print_summary(summary: &Vec<(PathBuf, bool)>, command_kind: CommandKind) {
+    println!("\nSummary for command {}:", command_kind);
+    let mut successes = 0;
+    for (path, success) in summary {
+        let (checkmark, status) = if *success {
+            ("[✓]", "succeeded")
+        } else {
+            ("[x]", "failed")
+        };
+        println!("  {}: {} {}!", checkmark, path.display(), status);
+        if *success {
+            successes += 1;
+        }
+    }
+    let failures = summary.len() - successes;
+    let successes_str = if successes == 1 {
+        "success"
+    } else {
+        "successes"
+    };
+    let failures_str = if failures == 1 { "failure" } else { "failures" };
+    println!(
+        "{} {}, {} {}",
+        successes, successes_str, failures, failures_str
+    );
+
+    if failures > 0 {
+        std::process::exit(1);
```

Comment:
Generally I think it's better to avoid hiding early program exits like this in seemingly unrelated functions. `print_summary` doesn't sound like the kind of function that could terminate the whole program. It might be better to return a result or whether or not it was successful so that `std::process::exit(1)` can be called from some higher level.